src/renderers/shared/utils/ReactErrorUtils.js

```diff
@@ -59,6 +63,7 @@ const ReactErrorUtils = {
   /**
    * Same as invokeGuardedCallback, but instead of returning an error, it stores
    * it in a global so it can be rethrown by `rethrowCaughtError` later.
+   * TODO: See if _caughtError and _rethrowError can be unified.
```

Comment:
By the way we could remove the boolean flag if we used a sentinel. E.g. `var noError = {}` and then `error === noError`.